### PR TITLE
Refactor material-ui out of stores portal 

### PR DESCRIPTION
**Pull request by Raghav-Maheshwari**

### CODE REVIEW

File: pages/_document.jsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Document, { Head, Main, NextScript } from 'next/document';
 import { SheetsRegistry, JssProvider } from 'react-jss';
 
 export default class JssDocument extends Document {
-  static getInitialProps(ctx) {
+  static async getInitialProps(ctx) {
```
Comment (QuinnNgo) [inline pages/_document.jsx]: why is this necessary? was this just broken before?

File: pages/stores_portal/[storeId].jsx [hunk 2]
```diff
@@ -1,28 +1,42 @@
...
       return item;
     });
 
-    return { items };
+    return { items, loggedIn };
+  }
+
+  componentDidMount() {
+    const { loggedIn } = this.props;
+    // check if cookie corresponds to current store:
```
Comment (QuinnNgo) [inline pages/stores_portal/[storeId].jsx]: this comment is a little confusing -- what is this for?

File: pages/stores_portal/[storeId].jsx [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
     const items = await res.json();
 
+    const loggedIn = cookies(ctx).loggedInStore === ctx.query.storeId;
+
     items.map(item => {
-      item.isChecked = false; // eslint-disable-line no-param-reassign
+      // eslint-disable-next-line no-param-reassign
+      item.isChecked = false;
       return item;
     });
```
Comment (QuinnNgo) [inline pages/stores_portal/[storeId].jsx]: best keep all the `items` code together:

```js
const itemsJSON = await res.json();
const items = itemsJSON.map(item => ({ ...item, isChecked: false });
```

this will also get rid of the lint error

File: pages/stores_portal/index.jsx [hunk 4]
```diff
@@ -132,46 +136,31 @@ class StoreLogin extends React.Component {
...
+            <div className={classes.invalid}>
+              There was an issue. If you cannot login, contact us at
+              hello@giveduet.org
+            </div>
+          )}
+          <div className={classes.button} onClick={this.handleSubmit}>
+            Login
           </div>
-        </form>
+        </div>
```
Comment (QuinnNgo) [inline pages/stores_portal/index.jsx]: If you switch this button to a div, you probably need to add `cursor: pointer` to its styles.

File: pages/stores_portal/[storeId].jsx [hunk 5]
```diff
@@ -1,28 +1,42 @@
...
+import Router, { withRouter } from 'next/router';
 import fetch from 'isomorphic-unfetch';
+import cookies from 'next-cookies';
 
 import Layout from '../../components/shared/layout';
 import Portal from '../../components/StorePortal';
 
 class StorePortal extends React.PureComponent {
-  static async getInitialProps({ query }) {
+  static async getInitialProps(ctx) {
```
Comment (QuinnNgo) [inline pages/stores_portal/[storeId].jsx]: any particular reason for changing this?

File: pages/_document.jsx (hunk 1 above)
Comment (Raghav-Maheshwari) [reply pages/_document.jsx]: ah good catch, its required elsewhere but not here

File: pages/stores_portal/[storeId].jsx (hunk 2 above)
Comment (Raghav-Maheshwari) [reply pages/stores_portal/[storeId].jsx]: so basically currently for the store portal we don't have a true "login" back-end. We basically just ask for a store email address and return the store object from our back-end, passing in the store ID as a query param `giveduet.org/stores_portal/[storeId]` once a store owner "logs in"

If someone wanted, they could just rename the url to pass in any store ID and thus access items from another store. I added cookies to basically track which store is currently logged in the session, and if they hit a route i.e `/stores_portal/2` even though their storeId is 1, then it'll redirect back to the login page.

File: pages/stores_portal/[storeId].jsx (hunk 5 above)
Comment (Raghav-Maheshwari) [reply pages/stores_portal/[storeId].jsx]: yeah the cookies library requires access to the `ctx` object

File: pages/stores_portal/[storeId].jsx (hunk 2 above)
Comment (QuinnNgo) [reply pages/stores_portal/[storeId].jsx]: Oh, I see. Is it possible for a person to be the admin of more than one store? And how do we know that `loggedIn` means that they were logged in to this particular store?

Comment (Raghav-Maheshwari) [reply pages/stores_portal/[storeId].jsx]:
```
// check if the storeId query param corresponds to the storeId stored in cookies: 
const loggedIn = cookies(ctx).loggedInStore === ctx.query.storeId;
```

Comment (Raghav-Maheshwari) [reply pages/stores_portal/[storeId].jsx]: and re admin of more than one store: currently we only work with store owners who operate one store location... but I think once we grow and potentially have admins of multiple store locations, we'll build a more robust store portal

Comment (QuinnNgo) [reply pages/stores_portal/[storeId].jsx]: Ah, i see. okay, this is fine for now since security isn't our main concern, but if a user is malicious, cookies still aren't a great way to handle this (anyone can change their cookies however they see fit). at some point, we'll want to do two things:

1. If every user can only correspond to one store, we can just have 
`giveduet.org/stores_portal` lead any user to their own store's information. You'd just make a route for `/me/store` that uses a user's auth info to grab their store.
2. regardless of whether each user corresponds to one store or many, this is best handled on the server. The endpoints for these stores should return `403 Unauthorized` if an invalid user makes a request to a store.
